e2e/specs/cancel-admission-request.spec.ts

```diff
@@ -0,0 +1,71 @@
+import { expect } from '@playwright/test';
+import { test } from '../core';
+import {
+  endVisit,
+  startVisit,
+  changeToWardLocation,
+  generateRandomPatient,
+  getProvider,
+  generateWardAdmission,
+  deletePatient,
+} from '../commands';
+import { type Visit } from '@openmrs/esm-framework';
+import { type Patient, type Encounter, type Provider, type Bed, type BedType } from '../types';
+import { deleteBed, deleteBedType, generateBedType, generateRandomBed } from '../commands/bed-operations';
+
+let visit: Visit;
+let wardPatient: Patient;
+let encounter: Encounter;
+let provider: Provider;
+let bed: Bed;
+let bedtype: BedType;
+
+test.beforeEach(async ({ api }) => {
+  await changeToWardLocation(api);
+  bedtype = await generateBedType(api);
+  bed = await generateRandomBed(api, bedtype);
+  provider = await getProvider(api);
+  wardPatient = await generateRandomPatient(api, process.env.E2E_WARD_LOCATION_UUID);
+  visit = await startVisit(api, wardPatient.uuid, process.env.E2E_WARD_LOCATION_UUID);
+  encounter = await generateWardAdmission(api, provider.uuid, wardPatient.uuid);
+});
+
+test('Cancelling an admission request', async ({ page }) => {
+  const fullName = wardPatient.person?.display;
+
+  await test.step('When I visit the patient ward page', async () => {
+    await page.goto(process.env.E2E_BASE_URL + `/spa/home/ward`);
+  });
+
+  await test.step('And I click the "Manage" button to view admission requests', async () => {
+    await page.getByRole('button', { name: 'Manage' }).click();
+  });
+
+  await test.step('Then I should see an admission request for the patient', async () => {
+    await expect(page.getByText(fullName)).toBeVisible();
+  });
+
+  await test.step('And when I click the "Cancel" button to cancel the request', async () => {
+    await page.getByRole('button', { name: 'Cancel' }).click();
+    await expect(page.getByText('Cancel admission request')).toBeVisible();
+    await expect(page.getByRole('heading', { name: 'Clinical notes' })).toBeVisible();
+  });
```

Comment:
```suggestion
  await test.step('And when I click the "Cancel" button to cancel the request', async () => {
    await page.getByRole('button', { name: 'Cancel' }).click();
  });
  
  await test.step('Then I should see the "Cancel admission request" form launched in the workspace", async() => {
  	 await expect(page.getByText('Cancel admission request')).toBeVisible();
  	 await expect(page.getByRole('heading', { name: 'Clinical notes' })).toBeVisible();
  });
```

Let's factor the workspace assertions into a separate step.